app/views/products/_form.html.haml

```diff
@@ -0,0 +1,29 @@
+%div.container
+  %div.row.justify-content-center
+    %div.col-10
+      =bootstrap_form_with model: @product, class: "shadow p-3 mb-3 bg-info rounded", local: true do |f|
+        =f.text_field :name, placeholder: "Name of product"
+        %div.row
+          %div.col
+            =f.file_field :image, accept: 'image/jpeg,image/jpg,image/png', id: "input"
+            -if @product.image.attached?
+              =link_to 'Delete image', active_storage_attachment_path(@product.image.id), method: :delete, remote: :true, id: "button", class: "btn btn-outline-light"
+            -else
+              %div.btn.btn-outline-light{ id: "button"} Delete image
+          %div.col.justify-content-center
+            -if @product.image.attached?
+              =image_tag @product.image, size: "500x200", id:"your_preview_id"
+            -else
+              %img{id: "your_preview_id"}
+
+        %div.row
+          %div.col
+            =f.text_field :code, placeholder: "Vendor code of product"
+          %div.col
+            =f.number_field :price, placeholder: "Price of product", label: "Price(USD)"
+        =f.text_area :description, placeholder: "Description of product"
+        =link_to 'Cancel', '/admin', class: "btn btn-outline-light btn-lg", data: { confirm: "Are you sure?" }
+        =f.submit class: "btn btn-outline-light btn-lg"
+
+=javascript_pack_tag 'delete_button'
+=javascript_pack_tag 'show_image'
```

Comment:
I'm not sure that it's a good idea to include this javascript inside this file, let's import it in the main application js file, but create a separate folder for this (app/javascript/packs/products)